pandas/core/algorithms.py

```diff
@@ -652,17 +651,13 @@ def factorize(values, sort=False, order=None, na_sentinel=-1, size_hint=None):
     original = values
 
     if is_extension_array_dtype(values):
-        values = getattr(values, "_values", values)
+        values = extract_array(values)
         labels, uniques = values.factorize(na_sentinel=na_sentinel)
         dtype = original.dtype
     else:
         values, dtype, _ = _ensure_data(values)
 
-        if (
-            is_datetime64_any_dtype(original)
-            or is_timedelta64_dtype(original)
-            or is_period_dtype(original)
-        ):
+        if original.dtype.kind in ["m", "M"]:
```

Comment:
Perhaps a silly question but wouldn't the datetimelike arrays pass through the `is_extension_array_dtype` branch here?

Reply:
the period and dt64tz go through that EA branch above, td64 and dt64-naive go through here

Reply:
can you add a comment to that effect. (can also merge this and do that later)

Reply:
will comment in next pass